packages/react-storage/src/components/StorageBrowser/views/LocationActionView/utils/getDestinationPickerDataTable.ts

```diff
@@ -0,0 +1,60 @@
+import { WithKey } from '../../../components/types';
+import { DataTableProps } from '../../../composables/DataTable';
+import { DataTableRow } from '../../../composables/DataTable/DataTable';
+
+const DESTINATION_PICKER_COLUMNS: DataTableProps['headers'] = [
+  { key: 'key', type: 'sort', content: { label: 'Folder name' } },
+];
+
+export const getDestinationListFullPrefix = (
+  destinationList: string[]
+): string => {
+  if (
+    destinationList.length < 1 ||
+    (destinationList.length === 1 && destinationList[0] === '')
+  )
+    return '';
+  // filter out root bucket ""
+  const destination = destinationList.filter((item) => item !== '').join('/');
+  return destination.endsWith('/') ? destination : `${destination}/`;
+};
```

Comment:
Non-blocking: Why doesn't the logic are we using for creating the breadcrumbs in `useNavigation` work here?

Reply:
That logic has changed a few times since I initially wrote this, so will have to revisit.